zeal: migrate to pkgs/by-name, use qt6

Jylhis opened the pull request:
Motivated by qt5.qtwebengine being marked vulnerable (#435067):
- Move package from pkgs/data/documentation to pkgs/by-name/ze/zeal
- Switch from Qt5 to Qt6 as the default
- Remove zeal-qt5 and zeal-qt6 variants in favor of single Qt6 version
- Add aliases for deprecated Qt-specific variants

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

fabianhjr commented on pkgs/top-level/aliases.nix:
```diff
@@ -2230,6 +2230,8 @@ mapAliases {
   z3_4_8 = throw "'z3_4_8' has been removed in favour of the latest version. Use 'z3'."; # Added 2025-05-18
   zabbix50 = throw "'zabbix50' has been removed, it would have reached its End of Life a few days after the release of NixOS 25.05. Consider upgrading to 'zabbix60' or 'zabbix70'.";
   zabbix64 = throw "'zabbix64' has been removed because it reached its End of Life. Consider upgrading to 'zabbix70'.";
+  zeal-qt5 = throw "zeal-qt5 has been replaced by zeal"; # Added 2025-08-31
+  zeal-qt6 = throw "zeal-qt6 has been replaced by zeal"; # Added 2025-08-31
```
It would be better to use a warn alias first

fabianhjr replied:
Example:

```
brasero-original = lib.warnOnInstantiate "Use 'brasero-unwrapped' instead of 'brasero-original'" brasero-unwrapped;
```

Jylhis replied:
I added the warning for `zeal-qt6` . For the qt5 version, is it preferred to keep it around for a while before removing it or should I keep the `throw` on that one?

fabianhjr replied:
I would prefer both being warns and backporting to 25.05 stable

Jylhis replied:
I can do that change, but just to double check that I understood correctly:

- Keep the `zeal-qt5` version in `all-packages.nix`, but point it to the `pkgs/by-name`
- and do the same type of warning as for the zeal-qt6

fabianhjr replied:
Was thinking more along the lines of auto-migrating:

```
zeal-qt5 = lib.warnOnInstantiate "'zeal-qt5' has been removed from nixpkgs. Please use 'zeal' instead" zeal; # Added 2025-08-31
```

Jylhis committed:
zeal: migrate to pkgs/by-name, use qt6

Motivated by qt5.qtwebengine being marked vulnerable (#435067):
- Move package from pkgs/data/documentation to pkgs/by-name/ze/zeal
- Switch from Qt5 to Qt6 as the default
- Remove zeal-qt5 and zeal-qt6 variants in favor of single Qt6 version
- Add aliases for deprecated Qt-specific variants
